test/javascript/turbolinks_visit_test.coffee

```diff
@@ -328,3 +328,12 @@ suite 'Turbolinks.visit()', ->
       assert.equal @window.pageYOffset, 42
       done()
     @Turbolinks.visit('iframe2.html', scroll: false)
+
+  test "doesn't scroll to top with :change on transition cache hit", (done) ->
```

Comment:
I would prefer a test that asserts that the page doesn't get restored from cache. The fact that it won't scroll to top is just a side effect of this (which already has a test [here](https://github.com/christophemaximin/turbolinks/blob/f491d2e2b78cc4434fd6b544080fcfe0ad2d8d62/test/javascript/turbolinks_visit_test.coffee#L316-L322)).